src/nvim/ops.c

```diff
@@ -3102,11 +3113,10 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
             curwin->w_cursor.col += (colnr_T)(totlen - 1);
           }
         }
-        if (VIsual_active)
+        if (VIsual_active) {
           lnum++;
-      } while (VIsual_active
-               && (lnum <= curbuf->b_visual.vi_end.lnum
-                   || lnum <= curbuf->b_visual.vi_start.lnum));
+        }
+      } while (VIsual_active && lnum <= end_lnum);
```

Comment:
Why not port 8.0.0225 first? It will be confusing to merge it later.

Reply:
Sorry, I didn't realize we had already covered this in #5781. Please try to mention that if you can, it will save time.